base-server/index.js

```diff
@@ -327,14 +333,24 @@ class BaseServer {
   }
 
   type (name, callbacks) {
-    if (typeof name === 'function') name = name.toString()
-    if (this.types[name]) {
-      throw new Error(`Action type ${name} was already defined`)
-    }
-    if (!callbacks || !callbacks.access) {
-      throw new Error(`Action type ${name} must have access callback`)
+    if (name instanceof RegExp) {
+      if (this.regexTypes[name]) {
+        throw new Error(`Action type ${name} was already defined`)
+      }
+      if (!callbacks || !callbacks.access) {
```

Comment:
It will be nice to avoid this check duplication. You can move this check before `if (name instanceof RegExp) {` test.